editor/src/components/canvas/multiplayer-cursors.tsx

```diff
@@ -0,0 +1,145 @@
+import React from 'react'
+import { Substores, useEditorState } from '../editor/store/store-hook'
+import { isLoggedIn } from '../editor/action-types'
+import { useMyPresence, useOthers } from '../../../liveblocks.config'
+import { motion } from 'framer-motion'
+import { multiplayerCursorColors } from './multiplayer'
+import { canvasPointToWindowPoint, windowToCanvasCoordinates } from './dom-lookup'
+import { windowPoint } from '../../core/shared/math-utils'
+import { UtopiaTheme, getPreferredColorScheme, useColorTheme } from '../../uuiui'
+import { safeIndex } from '../../core/shared/array-utils'
+
+export const MultiplayerCursors = React.memo(() => {
+  const colorTheme = useColorTheme()
+
+  const others = useOthers()
+  const [myPresence, setMyPresence] = useMyPresence()
+  const myColorIndex = React.useMemo(() => {
+    return myPresence.colorIndex ?? Math.floor(Math.random() * multiplayerCursorColors.light.length)
+  }, [myPresence])
+
+  const loginState = useEditorState(
+    Substores.userState,
+    (store) => store.userState.loginState,
+    'MultiplayerCursors loginState',
+  )
+  const canvasScale = useEditorState(
+    Substores.canvasOffset,
+    (store) => store.editor.canvas.scale,
+    'MultiplayerCursors canvasScale',
+  )
+  const canvasOffset = useEditorState(
+    Substores.canvasOffset,
+    (store) => store.editor.canvas.roundedCanvasOffset,
+    'MultiplayerCursors canvasOffset',
+  )
+
+  React.useEffect(() => {
+    if (!isLoggedIn(loginState)) {
+      return
+    }
+    const name = loginState.user.name ?? loginState.user.email ?? 'unknown'
```

Comment:
I'm not sure if it is a good idea to expose the email of a user (even as a fallback).